src/Microsoft.AspNet.Razor/RazorEditorParser.cs

```diff
@@ -12,45 +12,70 @@
+    /// details. However, the <see cref="PartialParseResult.Accepted"/> or <see cref="PartialParseResult.Rejected"/>
+    /// flags will ALWAYS be present.
+    /// </para>
+    /// <para>
+    /// A change can only be incrementally parsed if a single, unique, <see cref="Span"/> (see
+    /// <see cref="Parser.SyntaxTree"/>) in the syntax tree can be identified as owning the entire change.
+    /// For example, if a change overlaps with multiple <see cref="Span"/>s, the change cannot be parsed incrementally
+    /// and a full reparse is necessary. A <see cref="Span"/> "owns" a change if the change occurs either a) entirely
+    /// within it's boundaries or b) it is a pure insertion (see <see cref="TextChange"/>) at the end of a
+    /// <see cref="Span"/> whose <see cref="Span.EditHandler"/> can accept the change (see
+    /// <see cref="SpanEditHandler.CanAcceptChange"/>).
+    /// </para>
+    /// <para>
+    /// Even if a single unique <see cref="Span"/> owner can be identified, it's possible the edit will cause the
+    /// <see cref="Span"/> to split or merge with other <see cref="Span"/>s, in which case, a full reparse is necessary
+    /// to identify the extent of the changes to the tree.
+    /// </para>
+    /// <para>
+    /// When the <see cref="RazorEditorParser"/> returns <see cref="PartialParseResult.Accepted"/>, it updates
+    /// <see cref="CurrentParseTree"/> immediately. However, the editor is expected to update it's own data structures
+    /// independently. It can use <see cref="CurrentParseTree"/> to do this, as soon as the editor returns from
+    /// <see cref="CheckForStructureChanges"/>, but it should (ideally) have logic for doing so without needing the new
+    /// tree.
+    /// </para>
+    /// <para>
+    /// When <see cref="PartialParseResult.Rejected"/> is returned by <see cref="CheckForStructureChanges"/>, a
+    /// background parse task has <em>already</em> been started. When that task finishes, the
+    /// <see cref="DocumentParseComplete"/> event will be fired containing the new generated code, parse tree and a
+    /// reference to the original <see cref="TextChange"/> that caused the reparse, to allow the editor to resolve the
+    /// new tree against any changes made since calling <see cref="CheckForStructureChanges"/>.
+    /// </para>
+    /// <para>
+    /// If a call to <see cref="CheckForStructureChanges"/> occurs while a reparse is already in-progress, the reparse
+    /// is canceled IMMEDIATELY and <see cref="PartialParseResult.Rejected"/> is returned without attempting to
+    /// reparse. This means that if a consumer calls <see cref="CheckForStructureChanges"/>, which returns
+    /// <see cref="PartialParseResult.Rejected"/>, then calls it again before <see cref="DocumentParseComplete"/> is
+    /// fired, it will only receive one <see cref="DocumentParseComplete"/> event, for the second change.
+    /// </para>
```

Comment:
Reviewers: Please double-check these `<remarks>`. What was there before looked like it was copy-pasted from elsewhere and not updated since.